src/Data/Comp/Thunk.hs

```diff
@@ -80,7 +80,7 @@ whnf' = liftM (inject_ Inr) . whnf
 -- 'whnf' and then projects the top-level signature to the desired
 -- subsignature. Failure to do the projection is signalled as a
 -- failure in the monad.
-whnfPr :: (Monad m, g :<: f) => TermT m f -> m (g (TermT m f))
+whnfPr :: (MonadFail m, g :<: f) => TermT m f -> m (g (TermT m f))
```

Comment:
`MonadFail` isn't [introduced until `base=4.9.0.0`](http://hackage.haskell.org/package/base-4.12.0.0/docs/Control-Monad-Fail.html), so this patch breaks `4.7<=base<4.9`.  We need to up our minimum base version, either to `4.9`, or to `4.11`, when the import of `Control.Monad.Fail` [is redundant](https://wiki.haskell.org/MonadFail_Proposal).